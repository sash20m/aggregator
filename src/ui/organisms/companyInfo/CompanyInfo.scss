.company-content {
    display: flex;
    width: 100%;
    flex-direction: column;
}

.header {
    display: flex;
    width: 100%;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;

    &__avatar {
        width: 6.3em;
        margin-left: 80px;
        margin-top: 20px;
        margin-bottom: 20px;

        @media only screen and (max-width: 420px) {
            margin-left: 0;
        }

        @media only screen and (max-width: 585px) {
            margin-left: 0;
        }

    }

    &__info {
        margin-left: 20px;

        @media only screen and (max-width: 585px) {
            margin-left: 0;
        }

        &__name {
            font-size: 1.7rem;
            font-weight: 500;
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
            margin-bottom: 10px;

        }

        &__industry {
            font-size: 1.2em;
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
            margin-bottom: 10px;
            color: rgb(92, 92, 92);
        }
    }
}

.company-info {
    display: flex;
    width: 100%;
    background-color: rgb(241, 245, 246);
    align-items: center;
    flex-direction: column;
    height: 100vh;

    @media only screen and (max-width: 485px) {
        height: 112vh;
    }

    &__header-info {
        display: flex;
        width: 89%;
        flex-direction: row;
        margin-top: 30px;
        font-size: 1rem;
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
        box-shadow: 0 7px 27px 3px rgba(31, 105, 179, 0.05);

        @media only screen and (max-width: 700px) {
            flex-direction: column;
            width: 100%;
            align-items: center;
        }

        &__item {
            display: flex;
            flex-direction: column;
            width: 33.33%;
            height: 7em;
            background-color: white;
            justify-content: center;
            border: 1px solid #eaf0f6;

            @media only screen and (max-width: 700px) {
                width: 90%;
            }

            &__key {
                color: #416a8d;
                font-weight: 500;
                font-size: 0.9rem;
                margin: 0 0 10px 30px;
            }

            &__value {
                margin: 10px 0 0 30px;
                color: black;
                font-weight: 700;
            }

            &--first {
                display: flex;
                flex-direction: column;
                width: 33.33%;
                background-color: white;
                justify-content: center;
                border: 1px solid #eaf0f6;
                border-radius: 5px 0 0 5px;

                @media only screen and (max-width: 700px) {
                    width: 90%;
                    height: 7em;
                }
            }

            &--last {
                display: flex;
                flex-direction: column;
                width: 33.33%;
                background-color: white;
                justify-content: center;
                border: 1px solid #eaf0f6;
                border-radius: 0 5px 5px 0;

                @media only screen and (max-width: 700px) {
                    width: 90%;
                    height: 7em;
                }
            }
        }
    }

    &__footer-info {
        display: flex;
        width: 89%;
        flex-direction: row;
        margin-top: 30px;

        @media only screen and (max-width: 485px) {
            flex-direction: column;
            width: 90%;

        }

        &__contact {
            display: flex;
            flex-direction: column;
            width: 49%;
            background-color: white;
            border-radius: 5px;
            border: 1px solid #eaf0f6;

            @media only screen and (max-width: 485px) {
                width: 100%;
            }

            &__title {
                font-size: 1.2rem;
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                margin: 0;
                padding: 15px 0 15px 20px;
                border-bottom: 1px solid #eaf0f6;

                @media only screen and (max-width: 485px) {
                    width: 90%;
                }

            }

            &__row {
                display: flex;
                width: 100%;
                flex-direction: row;
                margin-top: 20px;

                @media only screen and (max-width: 1060px) {
                    flex-direction: column;
                }

                &--item {
                    margin-left: 20px;
                    margin-right: 200px;

                    @media only screen and (max-width: 1060px) {
                        margin-top: 10px;
                    }

                    &__key {
                        font-size: 1rem;
                        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                        color: rgb(1, 43, 63);
                        margin: 0;
                    }

                    &__value {
                        display: flex;
                        flex-direction: row;
                        font-size: 1rem;
                        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                        color: rgb(4, 63, 90);
                        margin: 0;

                        &__icon {
                            width: 20px;
                            height: 20px;
                            margin-left: 2px;
                            margin-right: 2px;
                        }

                        &__results {
                            display: flex;
                            flex-direction: column;

                            p {
                                margin: 0;
                            }
                        }
                    }


                }
            }

            &__column {
                display: flex;
                flex-direction: column;
                margin: 20px 0 20px 20px;

                &__key {
                    font-size: 1rem;
                    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                    color: rgb(1, 43, 63);
                    margin: 0;
                }

                &__value {
                    display: flex;
                    flex-direction: row;
                    font-size: 1rem;
                    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
                    color: rgb(4, 63, 90);
                    margin: 0;

                }
            }
        }

        &__map {
            display: flex;
            width: 49%;
            margin-left: 2%;
            height: 238px;
            justify-content: center;
            align-items: center;
            background-color: white;
            border-radius: 5px;
            border: 1px solid rgb(207, 207, 207);
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;

            @media only screen and (max-width: 485px) {
                margin: 20px 0 0 0;
                width: 100%;
            }

            &__google-map {
                width: 100%;
                height: 238px;
                border-radius: 5px;

            }
        }
    }

}