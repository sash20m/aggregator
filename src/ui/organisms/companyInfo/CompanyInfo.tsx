/* eslint-disable react/jsx-indent */
import React from 'react';

import { GoogleMap, LoadScript, Marker } from '@react-google-maps/api';
import { Company } from 'ui/templates/company/Company';

import './CompanyInfo.scss';

interface CompanyInfo {
  data: Company;
  imageURL: string;
}

interface Props {
  companyData: CompanyInfo;
}

export const CompanyInfo: React.FC<Props> = ({ companyData }) => {
  const { data, imageURL } = companyData;
  return (
    <div className="company-content" data-testid="company-content">
      <div className="header">
        <div className="header__content">
          {imageURL ? (
            <img
              className="header__avatar"
              src={`https://account.globaldatabase.com/logo/${imageURL}/`}
              alt=""
            />
          ) : (
            <div className="header__no-avatar">{data.name[0]}</div>
          )}

          <div className="header__info">
            <div className="header__info__name" data-testid="company-name">
              {data.name}
            </div>
            <div className="header__info__industry">{data.status.title}</div>
          </div>
        </div>
      </div>
      <div className="company-info">
        <div className="company-info__header-info">
          <div className="company-info__header-info__item--first">
            <p className="company-info__header-info__item__key">IDNO</p>
            <p className="company-info__header-info__item__value">
              {data.general_data.idno}
            </p>
          </div>
          <div className="company-info__header-info__item">
            <p className="company-info__header-info__item__key">
              Registration year
            </p>
            <p className="company-info__header-info__item__value">
              {data.general_data.creation_date}
            </p>
          </div>
          <div className="company-info__header-info__item--last">
            <p className="company-info__header-info__item__key">Staff</p>
            <p className="company-info__header-info__item__value">
              {data.general_data.size.name}
            </p>
          </div>
        </div>
        <div className="company-info__footer-info">
          <div className="company-info__footer-info__contact">
            <p className="company-info__footer-info__contact__title">
              Contact Information
            </p>
            <div className="company-info__footer-info__contact__row">
              <div className="company-info__footer-info__contact__row--item">
                <p className="company-info__footer-info__contact__row--item__key">
                  EMAIL:
                </p>
                <div className="company-info__footer-info__contact__row--item__value">
                  <img
                    className="company-info__footer-info__contact__row--item__value__icon"
                    src="../email.png"
                    alt="email"
                  />

                  <div className="company-info__footer-info__contact__row--item__value__results">
                    {data.general_data.contact_info.emails.length !== 0
                      ? data.general_data.contact_info.emails.map(
                          (item, key) => <p key={key}> Email</p>
                        )
                      : 'None'}
                  </div>
                </div>
              </div>
              <div className="company-info__footer-info__contact__row--item">
                <p className="company-info__footer-info__contact__row--item__key">
                  PHONE/FAX:
                </p>
                <div className="company-info__footer-info__contact__row--item__value">
                  <img
                    className="company-info__footer-info__contact__row--item__value__icon"
                    src="../phone.png"
                    alt="phone"
                  />

                  <div
                    className="company-info__footer-info__contact__row--item__value__results"
                    data-testid="phone"
                  >
                    {data.general_data.contact_info.phones.length !== 0
                      ? data.general_data.contact_info.phones.map(
                          (item, key) => <p key={key}>Phone</p>
                        )
                      : 'None'}
                  </div>
                </div>
                <div className="company-info__footer-info__contact__row--item__value">
                  <img
                    className="company-info__footer-info__contact__row--item__value__icon"
                    src="../mobile.png"
                    alt="mobile"
                  />
                  <div className="company-info__footer-info__contact__row--item__value__results">
                    {data.general_data.contact_info.mobile.length !== 0
                      ? data.general_data.contact_info.mobile.map(
                          (item, key) => <p key={key}>Mobile</p>
                        )
                      : 'None'}
                  </div>
                </div>
              </div>
            </div>
            <div className="company-info__footer-info__contact__column">
              <div className="company-info__footer-info__contact__column__key">
                {data.general_data.contact_info.sites[0] && (
                  <img
                    className="company-info__footer-info__contact__row--item__value__icon"
                    src="../www.png"
                    alt="mobile"
                  />
                )}
                Website
              </div>

              <p className="company-info__footer-info__contact__column__value">
                {data.general_data.contact_info.sites[0] ? imageURL : 'None'}
              </p>
            </div>
          </div>
          <div className="company-info__footer-info__map">
            {data.general_data.contact_info.address_de_facto.additional ? (
              <LoadScript googleMapsApiKey="AIzaSyCoiiNXqjXrJnG9d7mK3iRmy8ILIfJut2s">
                <GoogleMap
                  mapContainerClassName="company-info__footer-info__map__google-map"
                  center={{
                    lat:
                      data.general_data.contact_info.address_de_facto.additional
                        .lat,
                    lng:
                      data.general_data.contact_info.address_de_facto.additional
                        .long,
                  }}
                  zoom={17}
                >
                  <Marker
                    position={{
                      lat:
                        data.general_data.contact_info.address_de_facto
                          .additional.lat,
                      lng:
                        data.general_data.contact_info.address_de_facto
                          .additional.long,
                    }}
                    label={data.name}
                  />
                </GoogleMap>
              </LoadScript>
            ) : (
              <p>No Data for Map</p>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
