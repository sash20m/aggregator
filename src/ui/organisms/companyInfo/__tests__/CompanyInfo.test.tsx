import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import { CompanyInfo } from '../CompanyInfo';

const data = {
  name: 'Aridon',
  status: {
    title: 'Active',
  },
  general_data: {
    idno: '123456789',
    creation_date: 2020,
    size: {
      name: '10-20',
    },
    contact_info: {
      emails: [true],
      phones: [true, true],
      mobile: [true],
      address_de_facto: {
        additional: {
          lat: 20,
          long: 20,
        },
      },
      sites: ['www.aridon.com'],
    },
  },
};
const imageURL = 'www.aridon.com';

afterEach(cleanup);

describe('comapnyCard', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <CompanyInfo companyData={{ data, imageURL }} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be rendered', async () => {
    const { getByTestId } = render(
      <CompanyInfo companyData={{ data, imageURL }} />
    );

    expect(getByTestId('company-content')).toBeInTheDocument();
  });

  it('should receive company name trough props', async () => {
    const { getByTestId } = render(
      <CompanyInfo companyData={{ data, imageURL }} />
    );

    expect(getByTestId('company-name')).toHaveTextContent('Aridon');
  });

  it('should have 2 children', async () => {
    const { getByTestId } = render(
      <CompanyInfo companyData={{ data, imageURL }} />
    );

    expect(getByTestId('phone').childElementCount).toEqual(2);
  });
});
