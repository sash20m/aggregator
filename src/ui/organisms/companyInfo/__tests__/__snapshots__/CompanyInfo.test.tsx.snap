// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comapnyCard should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="company-content"
    data-testid="company-content"
  >
    <div
      class="header"
    >
      <div
        class="header__content"
      >
        <img
          alt=""
          class="header__avatar"
          src="https://account.globaldatabase.com/logo/www.aridon.com/"
        />
        <div
          class="header__info"
        >
          <div
            class="header__info__name"
            data-testid="company-name"
          >
            Aridon
          </div>
          <div
            class="header__info__industry"
          >
            Active
          </div>
        </div>
      </div>
    </div>
    <div
      class="company-info"
    >
      <div
        class="company-info__header-info"
      >
        <div
          class="company-info__header-info__item--first"
        >
          <p
            class="company-info__header-info__item__key"
          >
            IDNO
          </p>
          <p
            class="company-info__header-info__item__value"
          >
            123456789
          </p>
        </div>
        <div
          class="company-info__header-info__item"
        >
          <p
            class="company-info__header-info__item__key"
          >
            Registration year
          </p>
          <p
            class="company-info__header-info__item__value"
          >
            2020
          </p>
        </div>
        <div
          class="company-info__header-info__item--last"
        >
          <p
            class="company-info__header-info__item__key"
          >
            Staff
          </p>
          <p
            class="company-info__header-info__item__value"
          >
            10-20
          </p>
        </div>
      </div>
      <div
        class="company-info__footer-info"
      >
        <div
          class="company-info__footer-info__contact"
        >
          <p
            class="company-info__footer-info__contact__title"
          >
            Contact Information
          </p>
          <div
            class="company-info__footer-info__contact__row"
          >
            <div
              class="company-info__footer-info__contact__row--item"
            >
              <p
                class="company-info__footer-info__contact__row--item__key"
              >
                EMAIL:
              </p>
              <div
                class="company-info__footer-info__contact__row--item__value"
              >
                <img
                  alt="email"
                  class="company-info__footer-info__contact__row--item__value__icon"
                  src="../email.png"
                />
                <div
                  class="company-info__footer-info__contact__row--item__value__results"
                >
                  <p>
                     Email
                  </p>
                </div>
              </div>
            </div>
            <div
              class="company-info__footer-info__contact__row--item"
            >
              <p
                class="company-info__footer-info__contact__row--item__key"
              >
                PHONE/FAX:
              </p>
              <div
                class="company-info__footer-info__contact__row--item__value"
              >
                <img
                  alt="phone"
                  class="company-info__footer-info__contact__row--item__value__icon"
                  src="../phone.png"
                />
                <div
                  class="company-info__footer-info__contact__row--item__value__results"
                  data-testid="phone"
                >
                  <p>
                    Phone
                  </p>
                  <p>
                    Phone
                  </p>
                </div>
              </div>
              <div
                class="company-info__footer-info__contact__row--item__value"
              >
                <img
                  alt="mobile"
                  class="company-info__footer-info__contact__row--item__value__icon"
                  src="../mobile.png"
                />
                <div
                  class="company-info__footer-info__contact__row--item__value__results"
                >
                  <p>
                    Mobile
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="company-info__footer-info__contact__column"
          >
            <div
              class="company-info__footer-info__contact__column__key"
            >
              <img
                alt="mobile"
                class="company-info__footer-info__contact__row--item__value__icon"
                src="../www.png"
              />
              Website
            </div>
            <p
              class="company-info__footer-info__contact__column__value"
            >
              www.aridon.com
            </p>
          </div>
        </div>
        <div
          class="company-info__footer-info__map"
        >
          <div />
          <div>
            Loading...
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
